### ISSUE TITLE
How can I solve the attack of a large number of IPs?

### ISSUE THREAD

**Issue by gary81013** [SpigotMC/BungeeCord#3271]
How can I solve the attack of a large number of IPs?
![1](https://user-images.githubusercontent.com/43038433/158555315-6fcea24d-a12d-4026-a188-e7eb4a82fc73.PNG)
![f0081b4aaeff5dcc](https://user-images.githubusercontent.com/43038433/158555319-cbca4042-d5af-4c70-8587-c66813a07bce.PNG)

**Comment by Janmm14** [comment]
As long as there is no lag appearing I would suggest to ignore it.

**Comment by gary81013** [comment]
> 只要沒有出現滯後，我建議忽略它。

sometimes lag
Do I have a solution?

**Comment by Outfluencer** [comment]
Create a firewall with iptables or something like that

**Comment by gary81013** [comment]
> 使用 iptables 或類似的東西創建防火牆



> Create a firewall with iptables or something like that

Is there no way to solve it with a plugin?

**Comment by xIsm4** [comment]
> > 使用 iptables 或類似的東西創建防火牆
> 
> > Create a firewall with iptables or something like that
> 
> Is there no way to solve it with a plugin?

Yes obvuisly you can use an plugin that handles those type of connections.

**Comment by gary81013** [comment]
> > > 使用 iptables 或類似的東西創建防火牆
> > 
> > 
> > > Create a firewall with iptables or something like that
> > 
> > 
> > Is there no way to solve it with a plugin?
> 
> Yes obvuisly you can use an plugin that handles those type of connections.

can you tell me the plugin name

**Comment by gary81013** [comment]
I am willing to give $10 to someone who can help me solve my problem

**Comment by xIsm4** [comment]
> I am willing to give $10 to someone who can help me solve my problem

xIsm4 $avage#9127 write me on discord

**Comment by thefourcraft** [comment]
@gary81013 use Cloudflare to manage & control all the problems with your server
We use them if you own a big network of servers it's worth investing...

# Cloudflare setup guide
[Free Set up if you have port forward](https://www.youtube.com/watch?v=1HMHTQhuV9w&ab_channel=SoulStriker)
# Tunnel to secure your IP & server traffic 
[Ngrok Setup for extra security](https://www.youtube.com/watch?v=dC-hecmyxr4&ab_channel=ForeverJonny)
[cloudflared Tunnel - Enterprise plan recommended ](https://developers.cloudflare.com/cloudflare-one/connections/connect-apps/install-and-setup/)

**Comment by Janmm14** [comment]
> @gary81013 use Cloudflare to manage & control all the problems with your server We use them if you own a big network of servers it's worth investing...
> # Cloudflare setup guide
> 
> [Free Set up if you have port forward](https://www.youtube.com/watch?v=1HMHTQhuV9w&ab_channel=SoulStriker)
> # Tunnel to secure your IP & server traffic
> 
> [Ngrok Setup for extra security](https://www.youtube.com/watch?v=dC-hecmyxr4&ab_channel=ForeverJonny) [cloudflared Tunnel - Enterprise plan recommended ](https://developers.cloudflare.com/cloudflare-one/connections/connect-apps/install-and-setup/)

From what I know, this is the wrong cloudflare product. You need Cloduflare Spectrum and its very costly.

Instead I'd suggest tcpshield or manually search for open proxies and ban them via iptables for example.

**Comment by thefourcraft** [comment]
Well, yeah but if you are running a huge server operation then it's worth the cost...
most of the services that you need for one server network are free.
This is what they give for free 
![image](https://user-images.githubusercontent.com/90725541/159949247-fb129399-0497-43c0-8a06-9f3e837140f3.png)
![image](https://user-images.githubusercontent.com/90725541/159949301-4b71193f-3ded-42fb-8b83-c75d4393eb41.png)

The main thing that they prove for free and its **essential** is **security** 
and a lot of other cool things

**Comment by Janmm14** [comment]
> Well, yeah but if you are running a huge server operation then it's worth the cost... most of the services that you need for one server network are free. This is what they give for free ![image](https://user-images.githubusercontent.com/90725541/159949247-fb129399-0497-43c0-8a06-9f3e837140f3.png) ![image](https://user-images.githubusercontent.com/90725541/159949301-4b71193f-3ded-42fb-8b83-c75d4393eb41.png)
> 
> The main thing that they prove for free and its **essential** is **security** and a lot of other cool things

All those things apply only to web traffic. Only with cloudflare spectrum the minecraft traffic can be proxied and regulated by cloudflare.

**Comment by xIsm4** [comment]
> > Well, yeah but if you are running a huge server operation then it's worth the cost... most of the services that you need for one server network are free. This is what they give for free ![image](https://user-images.githubusercontent.com/90725541/159949247-fb129399-0497-43c0-8a06-9f3e837140f3.png) ![image](https://user-images.githubusercontent.com/90725541/159949301-4b71193f-3ded-42fb-8b83-c75d4393eb41.png)
> > The main thing that they prove for free and its **essential** is **security** and a lot of other cool things
> 
> All those things apply only to web traffic. Only with cloudflare spectrum the minecraft traffic can be proxied and regulated by cloudflare.

Yeah, but finaly got solved by me the attack.

**Comment by wolfdev1** [comment]
Cloudflare is the best option :s

**Comment by Janmm14** [comment]
> Cloudflare is the best option :s

no, cloudflare specturm too costly for minecraft ddos protection if you are not hypixel. other, cheaper alternatives exist.

**Comment by thefourcraft** [comment]
I have Cloudflare spectrum it's not that expensive ~500$ per month, but this also includes a lot of other products so I don't have an exact number

**Comment by Outfluencer** [comment]
No need for spektrum minecraft filter also is included in pro plan

**Comment by thefourcraft** [comment]
I don't think that this particular problem needs Cloudflare spectrum, I have a lot of servers for different games, 
not just Minecraft, and also all of our server infrastructure, so it makes sense to pay for good protection 
But Cloudflare is free for a lot of things and they give you essential protection which is kind of effective for kid's "DDoS" attacks

I don't think that an excellent high quality "someone" that can execute a DDoS attack to kill your server would be interested in a Minecraft server with ~50 players

**Comment by xIsm4** [comment]
> Cloudflare is the best option :s

He's server is cracked, wich means that **bots** are Legit connections and cloudfare won't do nothing against that attack, as jeanm said using a reverse proxy STACKED for Minecraft would be the best.
Any ways, he's using now my stuff
